### PR TITLE
[BOAT] Add roles in member log on member leave

### PR DESCRIPTION
**Pull request by BenSegal855**
This PR adds a list of roles a user had when they left. It also adds a `getMemberRoles` util function and updates the lookup command to use the new function.

### CODE REVIEW

File: boat/utils.js [hunk 1]
```diff
@@ -75,6 +75,14 @@ const utils = {
...
   plurialify (count, word) {
     return count === 1 ? word : `${word}s`
+  },
+
+  getMemberRoles (guild, member) {
+    const roles = []
+    guild.roles.filter(role => member.roles.includes(role.id)).forEach(role => {
+      roles.push(role.mention)
+    })
+    return roles
```
Comment (cyyynthia) [inline boat/utils.js]:
```suggestion
    return guild.roles.filter(role => member.roles.includes(role.id)).map(role => role.mention)
```

Comment (BenSegal855) [reply boat/utils.js]: Yes, but I just realized that I should really make that function return an array of the actual role objects and not of the mentions so Ill fix that now.

Comment (BenSegal855) [reply boat/utils.js]: Resolved by [b7b0c6d](https://github.com/powercord-org/powercord-backend/pull/53/commits/b7b0c6dd64685d40d24edc2307d7948608fbb79e)

File: boat/utils.js [hunk 2]
```diff
@@ -75,6 +75,10 @@ const utils = {
 
   plurialify (count, word) {
     return count === 1 ? word : `${word}s`
+  },
+
+  getMemberRoles (guild, member) {
+    return guild.roles.filter(role => member.roles.includes(role.id))
```
Comment (cyyynthia) [inline boat/utils.js]: I don't think it matters, but it'd be more efficient to map from member roles, which will always have less entries than there are server roles.
```suggestion
    return member.roles.map(id => guild.roles.get(id))
```

**Review by cyyynthia — APPROVED** [review]

**Pull request merged**